community/io/src/test/java/org/neo4j/io/pagecache/PageCacheTest.java

```diff
@@ -5210,4 +5211,244 @@ public void eagerFlushMustWriteToFileOnUnpin() throws Exception
+    }
+
+    @Test
+    public void noFaultNextReadMustStrideForwardMonotonically() throws Exception
+    {
+        configureStandardPageCache();
+        File file = file( "a" );
+        generateFileWithRecords( file, recordsPerFilePage * 6, recordSize );
+        try ( PagedFile pf = pageCache.map( file, filePageSize );
+              PageCursor faulter = pf.io( 0, PF_SHARED_READ_LOCK );
+              PageCursor nofault = pf.io( 0, PF_SHARED_READ_LOCK | PF_NO_FAULT ) )
+        {
+            assertTrue( faulter.next( 1 ) );
+            assertTrue( faulter.next( 3 ) );
+            assertTrue( faulter.next( 5 ) );
+            assertTrue( nofault.next() ); // Page id 0.
+            verifyNoFaultReadIsNotInMemory( nofault );
+            assertTrue( nofault.next() ); // Page id 1.
+            verifyNoFaultCursorIsInMemory( nofault, 1 );
+            assertTrue( nofault.next() ); // Page id 2.
+            verifyNoFaultReadIsNotInMemory( nofault );
+            assertTrue( nofault.next() ); // Page id 3.
+            verifyNoFaultCursorIsInMemory( nofault, 3 );
+            assertTrue( nofault.next() ); // Page id 4.
+            verifyNoFaultReadIsNotInMemory( nofault );
+            assertTrue( nofault.next() ); // Page id 5.
+            verifyNoFaultCursorIsInMemory( nofault, 5 );
+            assertFalse( nofault.next() ); // There's no page id 6..
+        }
+    }
+
+    @Test
+    public void noFaultReadCursorMustCopeWithPageEviction() throws Exception
+    {
+        configureStandardPageCache();
+        File file = file( "a" );
+        try ( PagedFile pf = pageCache.map( file, filePageSize );
+              PageCursor faulter = pf.io( 0, PF_SHARED_WRITE_LOCK );
+              PageCursor nofault = pf.io( 0, PF_SHARED_READ_LOCK | PF_NO_FAULT ) )
+        {
+            assertTrue( faulter.next() ); // Page id 0 now exists.
+            assertTrue( faulter.next() ); // And page id 1 now exists.
+            assertTrue( nofault.next() ); // No_FAULT cursor parked on page id 0.
+            verifyNoFaultCursorIsInMemory( nofault, 0 );
+            PageCursor[] writerArray = new PageCursor[maxPages - 1]; // The `- 1` to leave our `faulter` cursor.
+            for ( int i = 0; i < writerArray.length; i++ )
+            {
+                writerArray[i] = pf.io( 2 + i, PF_SHARED_WRITE_LOCK );
+                assertTrue( writerArray[i].next() );
+            }
+            // The page the nofault cursor is bound to should now be evicted.
+            for ( PageCursor writer : writerArray )
+            {
+                writer.close();
+            }
+            // If the page is evicted, then our read must have been inconsistent.
+            assertTrue( nofault.shouldRetry() );
+            // However, we are no longer in memory, because the page we had earlier got evicted.
+            verifyNoFaultReadIsNotInMemory( nofault );
+        }
+    }
+    // todo no fault read cursor must allow eviction of page
```

Comment:
todo

Reply:
👍 Removed it. It was a leftover reminder about the test just above.